e2e/terraform/provision-nomad/main.tf

```diff
@@ -1,28 +1,22 @@
 locals {
   provision_script = var.platform == "windows_amd64" ? "C:/opt/provision.ps1" : "/opt/provision.sh"
 
-  custom_path = dirname("${path.root}/config/custom/")
-
-  custom_config_files = compact(setunion(
-    fileset(local.custom_path, "nomad/*.hcl"),
-    fileset(local.custom_path, "nomad/${var.role}/*.hcl"),
-    fileset(local.custom_path, "nomad/${var.role}/indexed/*${var.index}.hcl"),
-    fileset(local.custom_path, "consul/*.json"),
-    fileset(local.custom_path, "consul/${var.role}/*.json"),
-    fileset(local.custom_path, "consul${var.role}indexed/*${var.index}*.json"),
-    fileset(local.custom_path, "vault/*.hcl"),
-    fileset(local.custom_path, "vault${var.role}*.hcl"),
-    fileset(local.custom_path, "vault${var.role}indexed/*${var.index}.hcl"),
+  config_path = dirname("${path.root}/config/")
+
+  config_files = compact(setunion(
+    fileset(local.config_path, "**"),
   ))
 
+  update_config_command = var.platform == "windows_amd64" ? "if (test-path /opt/config) { Remove-Item -Path /opt/config -Force -Recurse }; cp -r /tmp/config /opt/config" : "sudo rm -rf /opt/config; sudo mv /tmp/config /opt/config"
```

Comment:
Because there's some logic here and will probably grow more over time (see my comment below about change detection), I feel like it _might_ be better to add this to the `provision.ps1`/`provision.sh` script and have an argument we can call that only does configuration updates.

Reply:
I agree - I have tweaks that I'll apply in a follow up PR and will address this.